metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -126,28 +125,64 @@ final class MultilineStringFormattingProvider(
       textId: TextDocumentIdentifier,
       range: Range,
       newlineAdded: Boolean
-  )(
-      fn: (String, meta.Position) => List[TextEdit]
-  ): Future[List[TextEdit]] = Future {
-    val source = textId.getUri.toAbsolutePath
-    if (source.exists) {
-      val sourceText = buffer.get(source).getOrElse("")
-      val pos = range.getStart.toMeta(
-        Input.VirtualFile(source.toString(), sourceText)
-      )
-      if (pipeInScope(pos, sourceText, newlineAdded)) {
-        val tokens =
-          Input.VirtualFile(source.toString(), sourceText).tokenize.toOption
-        tokens.toList
-          .filter(multilineStringInTokens(_, pos, sourceText))
-          .flatMap(_ => fn(sourceText, pos))
+  )(fn: (String, meta.Position) => List[TextEdit]): Future[List[TextEdit]] =
+    Future {
+      val source = textId.getUri.toAbsolutePath
+      if (source.exists) {
+        val sourceText = buffer.get(source).getOrElse("")
+        val pos = range.getStart.toMeta(
+          Input.VirtualFile(source.toString(), sourceText)
+        )
+        if (pipeInScope(pos, sourceText, newlineAdded)) {
+          val tokens =
+            Input.VirtualFile(source.toString(), sourceText).tokenize.toOption
+          tokens.toList
+            .filter(multilineStringInTokens(_, pos, sourceText))
+            .flatMap(_ => fn(sourceText, pos))
+        } else Nil
       } else Nil
-    } else Nil
+    }
+
+  private def formatPipeLine(
+      line: Int,
+      lines: Array[String],
+      defaultIndent: String
+  ): TextEdit = {
+    val zeroPos = new Position(line, 0)
+    val lineText = lines(line)
+    val firstChar = lineText.trim.headOption
+
+    firstChar match {
+      case Some(char) =>
+        if (charIsPipe(char)) {
+          val firstPipeIndex = lineText.indexOf('|')
+          val firstCharAfterPipe = lineText.trim.tail.trim.headOption
+
+          firstCharAfterPipe match {
+            case Some(char) =>
```

Comment:
We could simplify it:
```
case Some(char) if charIsPipe(char) =>
   val secondPipeIndex = lineText.indexOf('|', firstPipeIndex + 1)
   val secondPipePos = new Position(line, secondPipeIndex)
   new TextEdit(new Range(zeroPos, secondPipePos), defaultIndent)
case _ =>
  val pipePos = new Position(line, firstPipeIndex)
  new TextEdit(new Range(zeroPos, pipePos), defaultIndent)
```

The same with the other match

Reply:
Ahh, yea, that is much nicer. I appreciate all the input!